orchagent/portsorch.cpp

```diff
@@ -2608,12 +2632,16 @@ bool PortsOrch::removeVlan(Port vlan)
         return false;
     }
 
-    sai_status_t status = sai_vlan_api->remove_vlan(vlan.m_vlan_info.vlan_oid);
-    if (status != SAI_STATUS_SUCCESS)
+    /* Do not delete default VLAN from driver, but clear internal state */
+    if (vlan.m_vlan_info.vlan_id != m_defaultVlan.m_vlan_info.vlan_id) 
```

Comment:
Move this check into the function `doVlanTask()` before `removeVlan()` function so that the this function's purpose is not touched. Add add a warning if the to-be-removed VLAN is the default VLAN, since it is not supposed to be removed.